pkg/registry/resource/resourceclaim/storage/metrics.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "kube_apiserver"
+	subsystem = "resourceclaim"
+)
+
+var (
+	// resourceClaimUpdateStatusDevicesAttempts tracks the number of
+	// ResourceClaims().Update calls (both successful and unsuccessful)
+	resourceClaimUpdateStatusDevicesAttempts = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "update_status_devices_attempts_total",
+			Help:           "Number of ResourceClaims update requests",
+			StabilityLevel: metrics.ALPHA,
+		})
+	// resourceClaimUpdateStatusDevicesFailures tracks the number of unsuccessful
+	// ResourceClaims().Update calls
```

Comment:
Do we need bespoke metrics for these operations **at all**? I think we get statistics from the apiserver for all resources and sub-resources, including status code as one dimension.

I missed it when the KEP was reviewed, otherwise I would have asked sooner, sorry!

Reply:
https://kubernetes.io/docs/reference/instrumentation/metrics/

> apiserver_request_total
> ...
> Labels:code component dry_run group resource scope subresource verb version

Reply:
I had the same question, but since was part of the criteria for the KEP and was not a big effort to implemented it I just moved with it

Reply:
I was also surprised there are other places in the pkg registry that add metrics, pod logs and certificates per example ...@johnbelamaric may be know better, but maybe this is part of PRR and is because it is easier to integrate

Reply:
Nope. We ask for metrics, if they are already available and they can tell us what they are, then that's fine. So, no need for specialized metrics if it's already covered. Just update the KEP to list the existing ones.

Reply:
@LionelJouin do you mind following up on updating the kep with the metrics @pohly is suggesting https://github.com/kubernetes/kubernetes/pull/130812#discussion_r1995858916

I'm going to close this since this is redundant

/close

Reply:
Thank you for the comments and suggestions, here is the update of the KEP: https://github.com/kubernetes/enhancements/pull/5206